www/js/app.js

```diff
@@ -961,12 +970,30 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
                     // It's an external URL : we should open it in a new tab
                     this.target = "_blank";
                 } else {
-                    // It's a link to another article
-                    // Add an onclick event to go to this article
+                    // It's a link to an article or file in the ZIM
+                    var decodedURL = decodeURIComponent(zimUrl);
+                    var contentType;
+                    // Some file types need to be downloaded rather than displayed (e.g. *.epub)
+                    // The HTML download attribute can be Boolean or a string representing the specified filename for saving the file
+                    // For Boolean values, getAttribute can return any of the following: download="" download="download" download="true"
+                    // So we need to test hasAttribute first: see https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute
+                    // However, we cannot rely on the download attribute having been set, so we also need to test for known download file types
+                    var downloadValue = this.hasAttribute('download') || regexpDownloadLinks.test(href);
+                    if (downloadValue) {
+                        downloadValue = this.getAttribute('download') || true;
+                        downloadValue = /download|true/i.test(downloadValue) || downloadValue;
```

Comment:
@mossroy Because the download attribute can be Boolean, or it can contain a string value representing the filename to use when saving the file in the filesystem, I've unified the two separate variables I had (downloadAttribute and downloadValue) into a single variable (downloadValue). I've tested this code with different values of the download attribute, and it seems to function as intended, i.e., if the attribute contains "", "download" or "true", it will be set to Boolean true (also true if there is no download attribute, but regexpDownloadLinks matches). But if the download attribute contains any other string value, it will be set to the string. However, I could only test it indirectly, as I can't manipulate the actual value of the attribute at debug time, and can only manipulate the variable itself (downloadValue). I'm 99% sure it works, but could you have a look and see if the code looks logically correct to you?

Reply:
I still find it complicated to read/understand.
You're right about the way to interpret the download attribute, and about the way to generate a file name. But maybe this code (that generates a filename) could be moved completely in parseAnchorsJQuery function, so that it then passes a filename (as a String) to goToArticle and displayFileDownloadAlert. It would simplify the understanding of the parameters of these functions (even if they were well documented).

Then, I would suggest to avoid reusing a variable for different purposes/types of content (except if it's really necessary, maybe for optimization).
For example, there could be a "isDownloadableLink" (boolean), then a "downloadAttributeValue" (string), and finally a "downloadFileName" (string). The first two ones are just temporary variables, used to generate the third one. As only the third one is used afterwards, this logic might even be externalized in a separate function, like "getDownloadFileName(this, decodedURL)", but maybe it's not necessary as there is not that much code to put in it.

Reply:
OK, I'll have another go!

Reply:
Thinking about this, I'm not keen on only passing a string in the `download` parameter of `goToArticle` because I think that function should remain generic in line with the HTMNL5 spec. In the spec, a download can be triggered either with a Boolean or with a filename. So it should be possible to trigger a download with just `download=true` passed to `goToArticle` . There might be other cases in the future where we want to trigger a download of content in line with the HTML5 spec. Given that, I think the only logical place to construct a filename if one is missing is in the appropriate function in `uiUtil.js`.

What I've done is to simplify the code in `parseAnchorsJQuery` as best I can following your advice above, except that I haven't included a `donwloadFileName` string here for the reasons stated. I have also moved the generation of a `contentType`, if it is missing from the HTML, into `uiUtil.js` so that we have consistent behaviour: if the download attribute is missing, we just pass a Boolean or null; if a `contentType` is missing, we just pass null. Then the download function in `uiUtil.js` takes care of constructing any missing data. I believe this is now better, and has decluttered `parseAnchorsJQuery`.

Let me know if you agree.

Reply:
All right, it can make sense.

Regarding the test `/download|true/i.test(downloadAttrValue)`, maybe it should be fixed for the case where downloadAttrValue contains one of those words without being equal to it.
I suppose you only want to match equality.
For example, if the download attribute is "truelle.zip", I suppose you want downloadAttrValue to be "truelle.zip" and not true.

Appart from that, I think you can squash/rebase/merge.

Thanks for this PR